src/ai/backend/storage/vast/vastdata_client.py

```diff
@@ -0,0 +1,440 @@
+    read_bw: int
+    read_iops: int
+    write_bw: float
+    write_iops: int
+
+
+def default_perf() -> Performance:
+    return {"read_bw": -1, "read_iops": -1, "write_bw": -1, "write_iops": -1}
+
+
+@dataclass(match_args=True)
+class VastClusterInfo:
+    id: str
+    guid: str
+    ip: str
+    mgmt_vip: str
+    name: str
+    state: str
+    physical_space: int = -1
+    physical_space_in_use: int = -1
+    usable_capacity_bytes: int = -1
+    iops: int = -1
+    rd_iops: int = -1
+    wr_iops: int = -1
+    bw: int = -1  # Bandwidth
+    rd_bw: int = -1
+    wr_bw: int = -1
+    latency: int = -1
+    rd_latency: int = -1
+    wr_latency: int = -1
+    max_file_size: int = -1
+    max_performance: Performance = field(default_factory=default_perf)
+
+    @classmethod
+    def from_json(cls, obj: Mapping[str, Any]) -> VastClusterInfo:
+        return VastClusterInfo(**{arg: obj.get(arg) for arg in cls.__match_args__})  # type: ignore[arg-type]
+
+
+@dataclass(match_args=True)
+class VastQuota:
+    path: str
+    id: VastQuotaID
+    guid: str
+    name: str
+    state: str
+    grace_period: str
+    soft_limit: int
+    hard_limit: int
+    soft_limit_inodes: int
+    hard_limit_inodes: int
+    used_inodes: int
+    used_capacity: int
+    percent_capacity: int
+
+    @classmethod
+    def from_json(cls, obj: Mapping[str, Any]) -> VastQuota:
+        return VastQuota(**{arg: obj.get(arg) for arg in cls.__match_args__})  # type: ignore[arg-type]
+
+
+class VastAPIClient:
+    api_endpoint: URL
+    api_version: APIVersion
+    username: str
+    password: str
+    ssl_context: ssl.SSLContext | bool | None
+    storage_base_dir: Path
+
+    _auth_token: TokenPair | None
+
+    def __init__(
+        self,
+        endpoint: str,
+        username: str,
+        password: str,
+        *,
+        api_version: APIVersion,
+        storage_base_dir: str,
+        ssl: ssl.SSLContext | bool | None = None,
+    ) -> None:
+        self.api_endpoint = URL(endpoint)
+        self.api_version = api_version
+        self.username = username
+        self.password = password
+        self.storage_base_dir = Path(storage_base_dir)
+
+        self._auth_token = None
+        self.ssl_context = ssl
+
+    @property
+    def _req_header(self) -> Mapping[str, str]:
+        assert self._auth_token is not None
+        return {
+            "Authorization": f"Bearer {self._auth_token['access_token']['token']}",
+            "Content-Type": "application/json",
+        }
+
+    async def _validate_token(self) -> None:
+        current_dt = datetime.now()
+        if self._auth_token is None:
+            return await self._login()
+        elif self._auth_token["access_token"]["issued_dt"] + DEFAULT_ACCESS_TOKEN_SPAN > current_dt:
+            return
+        elif (
+            self._auth_token["refresh_token"]["issued_dt"] + DEFAULT_REFRESH_TOKEN_SPAN > current_dt
+        ):
+            return await self._refresh()
+        return await self._login()
```

Comment:
VAST's API token is formed with JWT format so we can just evaluate `exp` value of JWT payload to determine if token is valid or not.